www/js/services.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+/* Notes on forecast.io's API:
+ *  - daily.data[0] is today
+ */
+
+weatherApp.constant('FORECASTIO_KEY', '23ecdce8d3b27ef562c7a1ccb5711b9f');
+
+/* weatherStore holds data for weatherData service.
+ */
+weatherApp.factory('weatherStore', function() {
+  var weatherStore = {
+    current: {}
+  };
+
+  return weatherStore;
+});
+
+
+
+/* weatherData service gets weather data (current, forecasts, and
+ * historical), processes it as necessary, and provides it to controllers.
+ */
+weatherApp.service('weatherData', ['$q', '$resource', '$http',
+  'FORECASTIO_KEY', 'weatherStore',
+  function($q, $resource, $http, FORECASTIO_KEY, weatherStore) {
+    this.getCurrentWeather = function(lat, lng) {
+      var url = 'https://api.forecast.io/forecast/' + FORECASTIO_KEY + "/" + lat + ',' + lng;
+      console.log(url);
+      // JSONP is only needed for "ionic serve".
+      // Simpler $http.get(url) works on devices.
+      return $http.jsonp(url + '?callback=JSON_CALLBACK').then(
+        function success(resp) {
+          weatherStore.current = resp.data;
+          console.log('GOT CURRENT');
+          console.dir(weatherStore.current);
+          return resp.data;
+        },
+        function failure(error) {
+          alert('Unable to get current conditions');
+          console.error(error);
+        });
+    };
+
+    this.getLocation = function() {
+      return $q(function(resolve, reject) {
+        navigator.geolocation.getCurrentPosition(function(position) {
+          resolve({
+            latitutde: position.coords.latitude,
+            longitude: position.coords.longitude
+          });
+        }, function(err) {
+          reject(err);
+        });
+      });
+    };
+
+    // TODO: move roundTemp into controller, since it is part of
+    // presentation, not weather data.
+
+    // Round temp to tenths of a degree.
+    this.roundTemp = function(temp) {
+      if (Math.abs(temp) >= 10) {
+        return temp.toPrecision(3);
+      } else {
+        return temp.toPrecision(2);
+      }
+    };
+
+    // Return current temperature
+    this.tempNow = function() {
+      return this.roundTemp(weatherStore.current.currently.temperature);
+    };
+
+    this.windSpeed = function() {
+      return this.roundTemp(weatherStore.current.currently.windSpeed);
+    };
+
+    this.humidity = function() {
+      return this.roundTemp(weatherStore.current.currently.humidity * 100);
+    };
+
+    this.visibility = function() {
+      return this.roundTemp(weatherStore.current.currently.visibility);
+    };
+
+    this.summary = function() {
+      return (weatherStore.current.currently.summary);
+    };
+
+
+    // Return tomorrow's high temperature.
+    this.tempTomorrowHigh = function() {
+      return this.roundTemp(weatherStore.current.daily.data[0].temperatureMax);
+    };
+
```

Comment:
This all looks good to me. Nicely spaced out and easy to read. I used $scope rather than _this_, and I wish I had used _this_ after reading this.